store/src/java/com/zimbra/cs/mime/Mime.java

```diff
@@ -1122,7 +1122,7 @@ public static String getFilename(MimePart mp) {
         // first, check the Content-Disposition header for the "filename" parameter
         try {
             String cdisp = mp.getHeader("Content-Disposition", null);
-            if (cdisp != null) {
+            if (cdisp != null && MimeConstants.CD_ATTACHMENT.equals(cdisp.trim())) {
```

Comment:
Please see commentary on the analysis ticket before moving forward with this approach.